plugins/outputs/cloudwatch/cloudwatch_test.go

```diff
@@ -323,45 +323,43 @@ func (svc *mockCloudWatchClient) PutMetricData(input *cloudwatch.PutMetricDataIn
 	return args.Get(0).(*cloudwatch.PutMetricDataOutput), args.Error(1)
 }
 
-func newCloudWatchClient(svc cloudwatchiface.CloudWatchAPI) *CloudWatch {
+func newCloudWatchClient(svc cloudwatchiface.CloudWatchAPI, forceFlushInterval time.Duration) *CloudWatch {
 	cloudwatch := &CloudWatch{
 		svc:                svc,
-		ForceFlushInterval: internal.Duration{Duration: 1 * time.Second},
+		ForceFlushInterval: internal.Duration{Duration: forceFlushInterval},
 	}
 	cloudwatch.startRoutines()
 	return cloudwatch
 }
 
-func TestWrite(t *testing.T) {
-	svc := new(mockCloudWatchClient)
-	res := cloudwatch.PutMetricDataOutput{}
-	svc.On("PutMetricData", mock.Anything).Return(
-		&res,
-		nil)
-	cloudWatchOutput := newCloudWatchClient(svc)
-	cloudWatchOutput.publisher, _ = publisher.NewPublisher(publisher.NewNonBlockingFifoQueue(10), 10, 2*time.Second, cloudWatchOutput.WriteToCloudWatch)
-	metrics := make([]telegraf.Metric, 0, 3)
+//
+func makeMetrics(count int) []telegraf.Metric {
+	metrics := make([]telegraf.Metric, 0, count)
 	measurement := "Test_namespace"
 	fields := map[string]interface{}{
 		"usage_user":       100,
-		"usage_system":     100,
-		"usage_idle":       100,
-		"usage_nice":       100,
-		"usage_iowait":     100,
-		"usage_irq":        100,
-		"usage_softirq":    100,
-		"usage_steal":      100,
-		"usage_guest":      100,
-		"usage_guest_nice": 100,
 	}
 
-	tags := map[string]string{"dimension_name1": "dimension_value2"}
```

Comment:
Any reason why to delete the dimensions?

Reply:
I don't think those were dimensions, just metric-names and metric-values.
I deleted it so that the the given `count` parameter is accurately the count of metrics.
Otherwise the input would have to be called `how_many_batches_of_10_metrics_would_you_like`. :)